sdks/java/io/hadoop-format/src/main/java/org/apache/beam/sdk/io/hadoop/format/HadoopFormatIO.java

```diff
@@ -437,6 +445,12 @@
           .build();
     }
 
+    /** Transforms the keys read from the source using the given key translation function. */
+    public Read<K, V> withKeyTranslation(SimpleFunction<?, K> function, Coder<K> coder) {
```

Comment:
You need to clear the keyCoder/valueCoder in the non coder based variants otherwise we won't honor the typedescriptor when the user changes the translation function

Reply:
Good catch. Fixed